README.md

```diff
@@ -243,3 +243,66 @@ A more generic Machine Learning DApp that illustrates how to use the [m2cgen (Mo
+
+While testing, we can easily forget to run the enviroment step, which causes the dapp not working properlly. In host mode running the enviroment part is the backend itself, so if it isn't executed it will not work.
+
+### Check if all files are generated for hostmode
+
+While building and creating an DApp, we can forget to generate all the files needed for it run in hostmode specially. Considering the production mode those files will be inside cartesi machine, those files need to be in the dapp folder in the host machine when we run this in hostmode. So, if your DApp generates binary files, model files and so on, be sure to generate those in the dapp folder too.
+
+### Disk space
+
+Be sure to have enough space to run the DApps properly. While building, it will be using many docker images of the Cartesi infrastructure so don't underestimate the disk usage. 
+
+### Free Docker from Sudo
+
+Some commands, specially while deploying an DApp, needs to use environment variables to work. If you, by some reason, run docker with `sudo`, it will not share the enviroment variables so it will cause the command to fail. So make sure to configure the host machine to not ask the sudo credentials while running docker.
+
+### Using wallets in the deploy process (Testnets)
+
+In the Deploy phase, we import a wallet through the `mnemonic` configured in the variables. In metamask we can have more than one wallet, but when you use an mnemonic you are using the first account. That said, be sure this account has the currency needed to deploy and DApp (The currency used in the desired testnet). If not, the deploy won't be sucessfull.
+
+### Importing a wallet to test the dapp in the localhost blockhain
+
+While testing an DApp, is natural to test it first in the localhost (hostmode) blockchain. But for that you will have to import one of the wallets listed in the first amount of the dapp logs. For that, you will have to use the private key listed there to import it to metamask. It can be done like below:
+
+- Choose one of the accounts in the list. 
+
+![image](https://user-images.githubusercontent.com/4421825/190483700-1c70cf8c-e975-43ba-886f-916d93e049b8.png)
+
+And then import it to metamask using the private key. If you sucessfully add one of the wallets to metamask your wallet shoud look like below. Remeber: you have to be running the blockchain while trying to add it.
+
+![image](https://user-images.githubusercontent.com/4421825/190484563-def28fde-775c-4b7b-a0ba-a8aa9112d02b.png)
+
+
+### Wallet can't connect to the blockchain in localhost
+
+Sometimes the transactions done with a localhost blockchain wallet will not complete due to various reasons. In those scenarios, you will have to re-import the wallet to metamask to force some synchronization with the localhost blockchain. Keep in mind that different DApps can generate different wallets in the wallet list shown at the start of running, so it can be also a situation that you have to use another wallet.
+
```

Comment:
On this topic "Wallet can't connect to the blockchain in localhost" we can add a warning telling that is necessary down the current docker composer and clear their volumes if user want running another dapp locally (in hostmode) and re-sign in wallet.